sdk/search/Azure.Search.Documents/src/Batching/SearchIndexingBufferedSenderOptions{T}.cs

```diff
@@ -0,0 +1,86 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading;
+using Azure.Search.Documents.Indexes;
+
+namespace Azure.Search.Documents
+{
+    /// <summary>
+    /// Provides the configuration options for
+    /// <see cref="SearchIndexingBufferedSender{T}"/>.
+    /// </summary>
+    /// <typeparam name="T">
+    /// The .NET type that maps to the index schema. Instances of this type
+    /// can be retrieved as documents from the index.
+    /// </typeparam>
+    public class SearchIndexingBufferedSenderOptions<T>
+    {
+        /// <summary>
+        /// Gets or sets a value indicating whether the sender should
+        /// automatically flush any indexing actions that have been added.
+        /// This will happen when the batch is full or when the
+        /// <see cref="AutoFlushInterval"/> has elapsed.  The default value is
+        /// <see langword="true"/>.
+        /// </summary>
+        public bool AutoFlush { get; set; } = true;
+
+        /// <summary>
+        /// Gets or sets an optional amount of time to wait before
+        /// automatically flushing any remaining indexing actions.  The default
+        /// value is 60 seconds.
+        /// </summary>
+        public TimeSpan? AutoFlushInterval
+        {
+            get => _autoFlushInterval;
+            set
+            {
+                // Treat any non-positive values as null.
```

Comment:
Shouldn't we throw? It seems very wrong to simply overwrite the value they set. If they provide an invalid value, either 1) it was intentionally and we should tell them otherwise, or 2) it was a problem in some calculation in their app and they probably want a stack to diagnose bad behavior.

Reply:
It feels like this is more of a coercion than a change.  We're preferring `null` to represent an infinite timeout, but `TimeSpan.FromSeconds(-1)` would work the same.  I'll overwrite it later though so we get the same behavior without changing their value.

Reply:
Moved the coercion deeper so customer values won't change out from under them.